### ISSUE TITLE
Updating `mesh.lines` produces inconsistent branching

### ISSUE THREAD

**Issue by glyg** [maartenbreddels/ipyvolume#77]
Here is a minimal example creating the code:

```python
import numpy as np
import ipyvolume as ipv
ipv.version_info
>>> (0, 4, 0, 'final', 0)
s = 1/2**0.5
# 4 vertices for the tetrahedron
x = np.array([1.,  -1, 0,  0])
y = np.array([0,   0, 1., -1])
z = np.array([-s, -s, s,  s])
colors = [(0., 0., 0.), (1., 0., 0.),
           (0., 1., 0.), (0., 0., 1.)]
# draw lines only from the first vertex
lines = [[0, 1], [0, 2], [0, 3]]


ipv.clear()
fig = ipv.figure()
# we draw the tetrahedron
ipv.plot_trisurf(x, y, z, lines=lines, color='black')
mesh = fig.meshes[0]
# and also mark the vertices
ipv.scatter(x, y, z, marker='sphere', color=colors)

ipv.xyzlim(-2, 2)
ipv.show()
```
This output this widget:
![ipyvolume-9](https://user-images.githubusercontent.com/1426279/31666363-35e99d82-b34c-11e7-91b0-948355672f2c.png)

The following should change connectivity from 4 lines going out from the black point
to two lines.
```python
mesh.lines = [[1, 2], [2, 3]] #that's red -> blue and blue -> green
fig
```
But outputs:

![ipyvolume-10](https://user-images.githubusercontent.com/1426279/31666373-41641458-b34c-11e7-9bc9-027bcb6493ae.png)

### Version infos 
Notebook version: 5.0.0
ipywidgets version: 7.0.1
ipyvolume version: 0.4.0

**Comment by maartenbreddels** [comment]
I hope it is fixed with 5fbaccbae886a4c3d1969e1f1928a1b5bfc9b496 let me know!

**Comment by maartenbreddels** [comment]
For the record, there was no need to show the figure again, the widget should update directly.

**Comment by glyg** [comment]
Hey, I tried the latest master (& bumped the version number locally to make sure that's what I'm running), but I still see the same behavior, sorry, unless I missed something when I updated or nbextensions gets in the way (??).

**Comment by maartenbreddels** [comment]
I usually run: `(cd js; ./node_modules/webpack/bin/webpack.js --watch)` to update the js, best is to then open an incognito window to make sure the cache doesn't do its work, or open the Chrome dev console and disable cache in the network tab (or whatever the equivalent is in your browser)

**Comment by glyg** [comment]
Ok I'll try that tomorrow!

Le 17 octobre 2017 20:25:56 GMT+02:00, Maarten Breddels <notifications@github.com> a écrit :
>I usually run: `(cd js; ./node_modules/webpack/bin/webpack.js --watch)`
>to update the js, best is to then open an incognito window to make sure
>the cache doesn't do its work, or open the Chrome dev console and
>disable cache in the network tab (or whatever the equivalent is in your
>browser)
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/maartenbreddels/ipyvolume/issues/77#issuecomment-337323777

-- 
Morphogénie Logiciels
12 rue Camoin Jeune,
13004 Marseille. 
+33 651 95 94 00

**Comment by glyg** [comment]
It works! 
In order to have the extension updated I had to:

```bash
$ jupyter nbextension disable ipyvolume
```

first, and then again

```bash
$ jupyter nbextension install --py --symlink --sys-prefix ipyvolume
$ jupyter nbextension enable ipyvolume --py --sys-prefix
```

Before that I uninstalled ipywidgets entirely, but I don't think it's necessary.

For the ellipsoid viewer, the update is fast enough, but as the branching changes, the smoothing between views creates stray lines across the object, that resorb once the view is stabilized. I remember seeing something about that in the issue.

Thanks again, your soft rocks!
